src/briefcase/integrations/file.py

```diff
@@ -18,21 +20,87 @@
 from briefcase.integrations.base import Tool, ToolCache
 
 
-class Download(Tool):
-    name = "download"
-    full_name = "Download"
+class File(Tool):
+    name = "file"
+    full_name = "File"
 
     @classmethod
-    def verify_install(cls, tools: ToolCache, **kwargs) -> Download:
-        """Make downloader available in tool cache."""
+    def verify_install(cls, tools: ToolCache, **kwargs) -> File:
+        """Make File available in tool cache."""
         # short circuit since already verified and available
-        if hasattr(tools, "download"):
-            return tools.download
+        if hasattr(tools, "file"):
+            return tools.file
 
-        tools.download = Download(tools=tools)
-        return tools.download
+        tools.file = File(tools=tools)
+        return tools.file
 
-    def file(self, url: str, download_path: Path, role: str | None = None) -> Path:
+    def is_archive(self, filename: str | os.PathLike) -> bool:
+        """Can a file be unpacked via `shutil.unpack_archive()`?
+
+        The evaluation is based purely on the name of the file. A more robust
+        implementation may actually interrogate the file itself.
+
+        Notably, the types of archives that can be unpacked will vary based on the types
+        of compression the current Python supports. So, for example, if LZMA is not
+        available, then tar.xz archives cannot be unpacked.
+
+        :param filename: path to file to evaluate as an archive
+        """
+        filename = Path(filename)
+        file_extensions = {
+            # captures extensions like .tar.gz, .tar.bz2, etc.
+            "".join(filename.suffixes[-2:]),
+            # as well as .tar, .zip, etc.
+            filename.suffix,
+        }
+        return not file_extensions.isdisjoint(self.supported_archive_extensions)
+
+    @property
+    def supported_archive_extensions(self) -> set[str]:
+        return {
+            extension
+            for archive_format in shutil.get_unpack_formats()
+            for extension in archive_format[1]
+        }
+
+    def unpack_archive(
+        self,
+        filename: str | os.PathLike,
+        extract_dir: str | os.PathLike,
+        **kwargs,
+    ):
+        """Unpack an archive file in to a destination directory.
+
+        :param filename: File path for the archive
+        :param extract_dir: Target file path for where to unpack archive
+        :param kwargs: additional arguments for shutil.unpack_archive
+        """
+        self.tools.shutil.unpack_archive(
+            filename=filename,
+            extract_dir=extract_dir,
+            **{
+                **self._unpack_archive_kwargs(filename),
+                **kwargs,
+            },
+        )
+
+    def _unpack_archive_kwargs(self, archive_path: str | os.PathLike) -> dict[str, str]:
```

Comment:
Possibly a little over-enthusiastic refactoring here; function calls aren't free, and in practice, this method can't be overwritten by a subclass. Given it's wrapping a single if clause that won't be needed in a couple of Python releases, I'm not sure the extra method is worth it.

Reply:
That makes sense. I also tweaked it a bit so (in theory, anyway) pyupgrade should have an easier time trimming it down later on.